src/sinks/s3_common/config.rs

```diff
@@ -312,7 +314,12 @@ impl RetryLogic for S3RetryLogic {
     type Response = S3Response;
 
     fn is_retriable_error(&self, error: &Self::Error) -> bool {
-        is_retriable_error(error)
+        let retry = is_retriable_error(error);
+        emit!(CheckRetryEvent {
```

Comment:
can we log the error with out decision to retry or not as well. so we know we got the error and its type etc. and our decision about. would help in improving  is_retriable_error